Added `location` and `history` to `MetaP`. (#238)
pals-project/pals: source/parameters/meta.md

(s:meta.params)=
## MetaP: Metadata Parameters

`MetaP` has six standard components:
```{code} yaml
MetaP:
  alias: ""         # [string] An alternate name for the element.
  ID: ""            # [string] An identification string.
  label: ""         # [string] A label string
  description: ""   # [string] A descriptive string

>   location: ""      # [string] To describe the location of an object.

Reviewer: An alternative docstring that follows the noun-phrase style of the other components and includes some examples to define what "location" means here (is this what "location" means here?):

```suggestion
  location: ""      # [string] Physical location of the object (e.g., building, room, or facility position).
```

Author: See my changes

>   history: ""       # [string] Object's history.

Reviewer: An alternative docstring that includes some examples to define what "history" means here (is this what "history" means here?):
```suggestion
  history: ""       # [string] Record of notable events in the object's lifecycle (e.g., manufacture, installation, maintenance).
```

Author: See my changes.

```
In addition to an element's `name`, these four strings can be used for pattern matching
when trying to locate all elements of a given type.

Besides the six standard strings, the `MetaP` parameter group can be used to store metadata 
that describes the lattice element but is not part of the PALS standard. 
Example metadata could be blueprint information, information on power supply connections, etc.

Components of `MetaP` are not limited to being simple strings or numbers but can be complex 
structures. The information contained in `MetaP` should be restricted to information that 

> does not affect simulations. Also program specific (as opposed to machine specific) information

Reviewer: I think I was erroneously parsing this "also" as "information that is additionally program-specific", but if that's not what this means here, maybe we could add a `,` to make it clearer for non-native speakers:

```suggestion
does not affect simulations. Also, program specific (as opposed to machine specific) information
```

(s:meta.params)=
## MetaP: Metadata Parameters

`MetaP` has six standard components:
```{code} yaml
MetaP:
  alias: ""         # [string] An alternate name for the element.
  ID: ""            # [string] An identification string.
  label: ""         # [string] A label string
  description: ""   # [string] A descriptive string

>   location: ""      # [string] To describe the location of element.

Reviewer: I would vote for style consistency with the `history` docstring here:

```suggestion
  location: ""      # [string] Element's location.
```

  history: ""       # [string] Element's history. 
```

> In addition to an element's `name`, these four strings can be used for pattern matching

Reviewer: Can `location` and `history` be used for pattern matching too?

```suggestion
In addition to an element's `name`, these six strings can be used for pattern matching
```

when trying to locate all elements of a given type.

Besides the six standard strings, the `MetaP` parameter group can be used to store metadata 
that describes the lattice element but is not part of the PALS standard. 
Example metadata could be blueprint information, information on power supply connections, etc.

Components of `MetaP` are not limited to being simple strings or numbers but can be complex 
structures. The information contained in `MetaP` should be restricted to information that 
does not affect simulations. Also, program specific (as opposed to machine specific) information
should be stored elsewhere.
```{code} yaml
quad1:                  # user-defined name

>   kind: Quadrupole      # [string] element switch

Reviewer: If `[string]` was removed intentionally from the docstring of `quad1` in this example, then maybe we should remove it from the docstring of `kind: Quadrupole` too?

```suggestion
  kind: Quadrupole      # element switch
```

Not sure what convention we follow in general (type reminder in examples or not).

  MetaP:
    ID: 0137-85
    location: East transfer line
    history:
      - 2022-04-01: Fixed water leak to vacuum.

>       - 2023-06-24: Fixed short in main bus bar.

Reviewer: In this example `history` is a list, not a string, while the YAML schema block above declares it as a string.

Author: Good catch! `history` (and two other components) should not have been marked as a string. I have revised the documentation.
